src/NzbDrone.Core/Download/Clients/Tribler/TriblerDownloadClient.cs

```diff
@@ -0,0 +1,319 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Http;
+using NzbDrone.Core.Configuration;
+using NLog;
+using FluentValidation.Results;
+using NzbDrone.Core.MediaFiles.TorrentInfo;
+using NzbDrone.Core.RemotePathMappings;
+using Tribler.Api;
+using System.Net.Http.Headers;
+using System.Collections.Generic;
+using System.Linq;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Parser.Model;
+using NzbDrone.Core.Validation;
+using MonoTorrent;
+
+namespace NzbDrone.Core.Download.Clients.Tribler
+{
+    using TriblerApi = global::Tribler.Api;
+
+    public class TriblerDownloadClient : TorrentClientBase<TriblerDownloadSettings>
+    {
+        private readonly Lazy<ITriblerApiClient> _triblerApiClient;
+
+        public TriblerDownloadClient(ITorrentFileInfoReader torrentFileInfoReader,
+                            IHttpClient httpClient,
+                            IConfigService configService,
+                            IDiskProvider diskProvider,
+                            IRemotePathMappingService remotePathMappingService,
+                            Logger logger)
+            : base(torrentFileInfoReader, httpClient, configService, diskProvider, remotePathMappingService, logger)
+        {
+            this._triblerApiClient = new Lazy<ITriblerApiClient>(CreateClient);
+
+        }
+
+        public override string Name => "Tribler";
+
+        public override bool PreferTorrentFile => false;
+
+        private System.Net.Http.HttpClient _httpClient;
+
+
+        protected ITriblerApiClient CreateClient()
+        {
+            _httpClient = new System.Net.Http.HttpClient();
```

Comment:
It doesn't look like this class uses `_httpClient` directly, so it can just be a local variable. Or better yet can we use the injected `_httpClient` as that handles a lot of things automatically. Though all the network requests here should be made through a secondary proxy class like the other download clients.

It may not be worthwhile using the auto generated code and instead using it as a guide to implement things in `TribblerProxy` using  Sonarr's `HttpClient`.

Reply:
I had hoped the OpenApi would have been easier to integrate.
Should use the same structure as the rest of the project now.